File: backend-go/internal/model/fund.go
package model

import "time"

// Fund 基金基础信息模型
type Fund struct {
	ID            string     `gorm:"type:uuid;primaryKey;default:uuid_generate_v4()" json:"id"`
	FundCode      string     `gorm:"type:varchar(16);uniqueIndex" json:"fund_code"`
	FundName      string     `gorm:"type:varchar(128)" json:"fund_name"`
	FundType      *string    `gorm:"type:varchar(32)" json:"fund_type,omitempty"`
	Manager       *string    `gorm:"type:varchar(64)" json:"manager,omitempty"`
	Custodian     *string    `gorm:"type:varchar(64)" json:"custodian,omitempty"`
	InceptionDate *time.Time `gorm:"type:date" json:"inception_date,omitempty"`
	Benchmark     *string    `gorm:"type:varchar(128)" json:"benchmark,omitempty"`
	Status        int16      `gorm:"default:1" json:"status"`
	CreatedAt     time.Time  `json:"created_at"`
	UpdatedAt     time.Time  `json:"updated_at"`
}

func (Fund) TableName() string { return "funds" }

File: ai-service-python/app/models/fund.py
from datetime import date

from sqlalchemy import String, SmallInteger, Date
from sqlalchemy.orm import Mapped, mapped_column

from app.models.base import Base, UUIDMixin, TimestampMixin


class Fund(UUIDMixin, TimestampMixin, Base):
    __tablename__ = "funds"

    fund_code: Mapped[str] = mapped_column(String(16), unique=True, nullable=False)
    fund_name: Mapped[str] = mapped_column(String(128), nullable=False)
    fund_type: Mapped[str | None] = mapped_column(String(32))
    manager: Mapped[str | None] = mapped_column(String(64))
    custodian: Mapped[str | None] = mapped_column(String(64))
    inception_date: Mapped[date | None] = mapped_column(Date)
    benchmark: Mapped[str | None] = mapped_column(String(128))
    status: Mapped[int] = mapped_column(SmallInteger, default=1)
